Sources/_StringProcessing/Algorithms/Consumers/RegexConsumer.swift

```diff
@@ -9,56 +9,54 @@
 //
 //===----------------------------------------------------------------------===//
 
-import _MatchingEngine
+public struct RegexConsumer<
```

Comment:
This looks fine to me for now.

Outside scope of this PR, we might make RegexProtocol itself a consumer/searcher/matcher. But, I think it's best to establish parent-child relationships after we have the fundamental model in place. The important thing is getting this code and conformances in.

BTW, why does this need to be public?

Reply:
Some consumer/searcher algorithms return a type generic over the searcher, so the regex overloads return a type generic over `RegexConsumer`. So same story as `PredicateConsumer` at the moment.